graylog2-web-interface/src/views/components/visualizations/hooks/useExtendedChartGeneratorSettings.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import { useCallback, useMemo } from 'react';
+
+import useFeature from 'hooks/useFeature';
+import useWidgetUnits from 'views/components/visualizations/hooks/useWidgetUnits';
+import type AggregationWidgetConfig from 'views/logic/aggregationbuilder/AggregationWidgetConfig';
+import {
+  generateMappersForYAxis,
+  getBarChartTraceOffsetSettings, getHoverTemplateSettings,
+} from 'views/components/visualizations/utils/chartLayoytGenerators';
+import type { BarMode } from 'views/logic/aggregationbuilder/visualizations/BarVisualizationConfig';
+import convertDataToBaseUnit from 'views/components/visualizations/utils/convertDataToBaseUnit';
+import type { AbsoluteTimeRange } from 'views/logic/queries/Query';
+import { NO_FIELD_NAME_SERIES, UNIT_FEATURE_FLAG } from 'views/components/visualizations/Constants';
+import { getBaseUnit } from 'views/components/visualizations/utils/unitConvertors';
+import FieldUnit from 'views/logic/aggregationbuilder/FieldUnit';
+import isLayoutRequiresBaseUnit from 'views/components/visualizations/utils/isLayoutRequiresBaseUnit';
+import getFieldNameFromTrace from 'views/components/visualizations/utils/getFieldNameFromTrace';
+import type { ChartDefinition } from 'views/components/visualizations/ChartData';
+
+const useExtendedChartGeneratorSettings = ({ config, barmode, effectiveTimerange }: {
```

Comment:
I think the naming is not great. From my understanding, this is applying custom unit formats and generates multiple axis/transforms values accordingly. This should be reflected in the naming.